src/main/java/ch/njol/skript/expressions/ExprStringCase.java

```diff
@@ -120,12 +160,58 @@ public boolean isSingle() {
 
 	@Override
 	public String toString(@Nullable Event e, boolean debug) {
-		if (literal != null)
-			return changeCase(literal, mode);
-		else if (origin != null && e != null)
-			return changeCase(origin.getSingle(e), mode);
-		
-		return "";
+		switch (type) {
+			case 0: // Basic Case Change 
+				return (casemode == 1) ? "uppercase" : "lowercase";
+			case 1: // Proper Case 
+				return ((casemode == 3) ? "strict" : "lenient") + " proper case";
+			case 2: // Camel Case 
+				return ((casemode == 3) ? "strict" : "lenient") + " camel case";
+			case 3: // Pascal Case 
+				return ((casemode == 3) ? "strict" : "lenient") + " pascal case";
+			case 4: // Snake Case 
+				return ((casemode == 0) ? "" : ((casemode == 1)) ? "upper " : "lower ") + "snake case";
+			case 5: // Kebab Case 
+				return ((casemode == 0) ? "" : ((casemode == 1)) ? "upper " : "lower ") + "kebab case";
+		}
+		return ""; // Shouldn't reach here anyways 
+	}
+	
+	@SuppressWarnings("null")
+	private static String toCamelCase(String str, boolean strict) {
+		String[] words = str.split(" "); // Splits at spaces 
+		String buf = words.length > 0 ? (strict ? words[0].toLowerCase() : WordUtils.uncapitalize(words[0])) : "";
+		for (int i = 1; i < words.length; i++)
+			buf += strict ? WordUtils.capitalizeFully(words[i]) : WordUtils.capitalize(words[i]);
+		return buf;
+	}
+	
+	private static String toPascalCase(String str, boolean strict) {
+		String[] words = str.split(" "); // Splits at spaces 
+		String buf = "";
+		for (int i = 0; i < words.length; i++)
+			buf += strict ? WordUtils.capitalizeFully(words[i]) : WordUtils.capitalize(words[i]);
+		return buf;
+	}
+	
+	@SuppressWarnings("null")
+	private static String toSnakeCase(String str, int mode) {
+		if (mode == 0)
+			return str.replace(' ', '_');
+		char[] arr = str.toCharArray();
```

Comment:
This doesn't support Unicode codepoints above 65536. Please iterate over codepoints instead.